File: weeks/01/3_breaking_the_records/breaking_the_records.rb
def breaking_records(scores)
    min_score = scores[0]
    max_score = scores[0]
    min_score_broken = 0
    max_score_broken = 0

    scores[1..-1].each do |score|
        # New max score
        if score > max_score
            max_score = score
            max_score_broken += 1
        end

        # New min score
        if score < min_score
            min_score = score
            min_score_broken += 1
        end
    end

    [max_score_broken, min_score_broken]
end

File: weeks/01/3_breaking_the_records/breaking_the_records.js
function breakingRecords(scores) {
    let minScore = scores[0];
    let maxScore = scores[0];
    let minScoreBroken = 0;
    let maxScoreBroken = 0;

    for (let i = 1; i < scores.length; i++) {
        let score = scores[i];

        // New max score
        if (score > maxScore) {
            maxScore = score;
            maxScoreBroken++;
        }

        // New min score
        if (score < minScore) {
            minScore = score;
            minScoreBroken++;
        }
    }

    return [maxScoreBroken, minScoreBroken];
}